main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -58,6 +394,69 @@ object Inliner extends Phase[SimplifiedAst.Root, SimplifiedAst.Root] {
     * - A UserError, MatchError, or SwitchError is worth zero points.
     * - An if-then-else statement is worth the value of the condition plus two times the sum of the consequent and alternative.
     */
-  def score(defn: Constant): Boolean = ??? // TODO
+  def score(definitions: Constant): Int = exprScore(definitions.exp)
+
+  /**
+    * Returns the score of the given expression `exp`.
+    */
+  def exprScore(exp: Expression): Int = exprListScore(0,List(exp))
+
+  /**
+    * Returns the total score of the list of expressions `exps`.
+    */
+  @tailrec def exprListScore(acc: Int, expressions: List[Expression]): Int =
```

Comment:
I understand why you put the `@tailrec`, but it is not necessary. A program might contain thousands and thousands of *definitions* but each definition is likely to be very small, i.e. expressions themselves are typically verify small trees. Hence I would prefer a simple straightforward recursion on the tree.